src/effects/VST3/VST3Effect.cpp

```diff
@@ -331,82 +331,68 @@ bool VST3Effect::SupportsRealtime() const
 
 bool VST3Effect::SupportsAutomation() const
 {
-   if(mEditController == nullptr)
-      return false;
-
    using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+
+   return AnyOf
+   (
+      [&](const ParameterInfo& parameterInfo)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
-            return true;
+         return (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate);
       }
-   }
-
-   return false;
+   );
 }
 
 bool VST3Effect::SaveSettings(
-   const EffectSettings &, CommandParameters & parms) const
+   const EffectSettings& settings, CommandParameters & parms) const
 {
-   if(mEditController == nullptr)
+   if (mEditController == nullptr)
       return false;
 
+   const VST3EffectSettings& ownSettings = GetSettings(settings);
+
    using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+
+   return ForEachParameter
+   (
+      [&](const ParameterInfo& parameterInfo)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
          {
             parms.Write(
                VST3Utils::MakeAutomationParameterKey(parameterInfo),
-               mEditController->getParamNormalized(parameterInfo.id)
+               ownSettings.mValues.at(parameterInfo.id)
```

Comment:
`at` might allow brevity but it can throw an exception that isn't handled anywhere.  I would rather use `find` and a test against `end` and skip parameters that are not found.